util/release/manifest.go

```diff
@@ -35,45 +35,54 @@ func manifest(args *docopt.Args) {
 		src = f
 	}
 
-	var lookup idLookupFunc
-	var err error
-	if file := args.String["--id-file"]; file != "" {
-		lookup, err = fileLookupFunc(file)
-	} else {
-		lookup, err = dockerLookupFunc()
-	}
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if err := interpolateManifest(lookup, args.String["--image-repository"], src, dest); err != nil {
+	if err := interpolateManifest(args.String["--image-dir"], args.String["--image-repository"], src, dest); err != nil {
 		log.Fatal(err)
 	}
 }
 
-var imageIDPattern = regexp.MustCompile(`\$image_id\[[^\]]+\]`)
+var imageArtifactPattern = regexp.MustCompile(`\$image_artifact\[[^\]]+\]`)
 
-func interpolateManifest(lookup idLookupFunc, imageRepository string, src io.Reader, dest io.Writer) error {
+func interpolateManifest(imageDir, imageRepository string, src io.Reader, dest io.Writer) error {
 	manifest, err := ioutil.ReadAll(src)
 	if err != nil {
 		return err
 	}
-	manifest = bytes.Replace(manifest, []byte("$image_repository"), []byte(imageRepository), -1)
 	var replaceErr interface{}
 	func() {
 		defer func() {
 			replaceErr = recover()
 		}()
-		manifest = imageIDPattern.ReplaceAllFunc(manifest, func(raw []byte) []byte {
-			imageName := string(raw[10 : len(raw)-1])
-			if !strings.Contains(imageName, "/") {
-				imageName = "flynn/" + imageName
+		manifest = imageArtifactPattern.ReplaceAllFunc(manifest, func(raw []byte) []byte {
+			name := string(raw[16 : len(raw)-1])
+
+			f, err := os.Open(filepath.Join(imageDir, name+".json"))
+			if err != nil {
+				panic(err)
+			}
+			defer f.Close()
+
+			image := &ct.ImageManifest{}
+			if err := json.NewDecoder(f).Decode(image); err != nil {
+				panic(err)
+			}
+
+			artifact := &ct.Artifact{
+				Type:     ct.ArtifactTypeFlynn,
+				URI:      fmt.Sprintf("%s?target=/images/%s.json", imageRepository, image.ID()),
+				Manifest: image,
+				Hashes:   image.Hashes(),
+				Meta:     map[string]string{"flynn.component": name},
+			}
+			if version.Dev() {
+				artifact.LayerURLTemplate = "file:///var/lib/flynn/layer-cache/{id}.squashfs"
+			} else {
+				artifact.LayerURLTemplate = fmt.Sprintf("%s?target=/layers/{id}.squashfs", imageRepository)
```

Comment:
I decided not to put the version in the URLs because there doesn't seem to be a need to. There will still be an image manifest per version (e.g. `/vXXX/images.json`) but it will just have URLs to images like `/image/SHA.json` containing layers with URLs like `/layer/SHA.layer`.